numpy/__init__.pyi

```diff
@@ -778,6 +779,21 @@ class nditer:
         buffersize: Any = ...,
     ) -> Any: ...
     def __getattr__(self, key: str) -> Any: ...
+    def __enter__(self) -> nditer: ...
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_value: Optional[BaseException],
+        traceback: Optional[TracebackType]
```

Comment:
```suggestion
        exc_type: None | Type[BaseException],
        exc_value: None | BaseException,
        traceback: None | TracebackType,
```
Style nit

Reply:
I agree for Union not so much  for Optional but I will make the change. Since PEP604 is 3.10 I would not have though to use it but I guess the constraint is lighter in type hints.

Reply:
> Since PEP604 is 3.10 I would not have though to use it but I guess the constraint is lighter in type hints.

Yup, the `|` syntax has been supported for annotations (not type aliases) since the [mypy 0.800 release](https://mypy-lang.blogspot.com/2021/01/mypy-0800-released.html).